detail.py

```diff
@@ -0,0 +1,47 @@
+from django_mako_plus import view_function, jscontext
+from catalog import models as cmod
+
+@view_function
+
+def process_request(request, product:cmod.Product):
+
+  # products = cmod.Product.objects.filter(id= request.dmp.urlparams[0])
+    if (product in request.LastFiveList):
+        request.LastFiveList.remove(product)
+    request.LastFiveList.insert(0,product)
+
+    if product.__class__.__name__ == 'BulkProduct':

```

Comment:
This seems like a lot of unnecessary code. I would see if there is a way that you can pass product differently to avoid of these if statements

Reply:
Okay. I will look into this. Do you have any suggestions as to where to start?